docs/specs/shapes.md

```diff
@@ -7,6 +7,52 @@ The graphical elements are divided in 4 categories:
 * [Styles](#shape-style), that define the visual appearance of shapes
 * [Modifiers](#modifier) alter the curves of the shapes
 
+## Shape Rendering Model
+
+### Grouping and Ordering Rules
+
+* **shapes** are rendered in reverse order (bottom->top)
+* **groups** offer a scoping mechanism for transforms, styles, modifiers, and shapes
```

Comment:
I think it would be helpful to more thoroughly define scope and it's contents here, as that concept is heavily used later. ie something like "all children of a group including sub-groups and their children are considered a part of the group's scope.